integration-tests/actions/vrfv2plus/vrfv2plus_steps.go

```diff
@@ -140,3 +156,101 @@ func FundVRFCoordinatorV2PlusSubscription(linkToken contracts.LinkToken, coordin
 	}
 	return chainClient.WaitForEvents()
 }
+
+func SetupVRFV2PlusEnvironment(
+	env *test_env.CLClusterTestEnv,
+	linkAddress contracts.LinkToken,
+	mockETHLinkFeedAddress contracts.MockETHLINKFeed,
+	isNativePayment bool,
+) (*test_env.CLClusterTestEnv, *VRFV2PlusContracts, *big.Int, *VRFV2PlusJobInfo, error) {
+
+	vrfv2PlusContracts, err := DeployVRFV2PlusContracts(env.ContractDeployer, env.EVMClient)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrDeployVRFV2PlusContracts)
+	}
+
+	err = env.EVMClient.WaitForEvents()
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrWaitTXsComplete)
+	}
+
+	err = vrfv2PlusContracts.Coordinator.SetConfig(
+		vrfv2plus_constants.MinimumConfirmations,
+		vrfv2plus_constants.MaxGasLimitVRFCoordinatorConfig,
+		vrfv2plus_constants.StalenessSeconds,
+		vrfv2plus_constants.GasAfterPaymentCalculation,
+		vrfv2plus_constants.LinkEthFeedResponse,
+		vrfv2plus_constants.VRFCoordinatorV2PlusFeeConfig,
+	)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrSetVRFCoordinatorConfig)
+	}
+	err = env.EVMClient.WaitForEvents()
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrWaitTXsComplete)
+	}
+
+	err = vrfv2PlusContracts.Coordinator.CreateSubscription()
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrCreateVRFSubscription)
+	}
+	err = env.EVMClient.WaitForEvents()
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrWaitTXsComplete)
+	}
+
+	subID, err := vrfv2PlusContracts.Coordinator.FindSubscriptionID()
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrFindSubID)
+	}
+
+	err = vrfv2PlusContracts.Coordinator.AddConsumer(subID, vrfv2PlusContracts.LoadTestConsumer.Address())
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, ErrAddConsumerToSub)
+	}
+
+	if isNativePayment {
```

Comment:
Is this `if` needed? Why not just fund both native + link in every setup? (and also set LINK/LINKETH feed, etc.)

Reply:
yeah, I guess you are right.
Just wanted to keep the setup as minimal as possible and to check how it will perform if we configure link and native token separately. 
But this will not happen in PROD, will change